src/allpairs.py

```diff
@@ -83,9 +85,12 @@ def run(nodetypes=None, network=None, output_file=None, singularity=False):
 
         print('Running All Pairs Shortest Paths with arguments: {}'.format(' '.join(command)), flush=True)
 
-        container_framework = 'singularity' if singularity else 'docker'
-        out = run_container(container_framework,
-                            'reedcompbio/allpairs',
+        container_framework = config.config.framework
```

Comment:
I struggled a bit with how to make that decision, too. Here was why I settled on grabbing the config here instead of at the very last moment in `run_container`:

In theory, this approach to creating a globally-accessible config object makes it a bit easier to implement fine-grained configurations at the PRM level. For example, we could easily adapt things to allow for the specification of per-PRM container frameworks. By keeping the container framework as a variable in each PRM's `run` method and passing it to `run_container`, we minimize future work to make each PRM aware of what framework it should be using. If that variable is grabbed at the last second in `run_container`, all PRMs are locked into using a global configuration.